interactions/api/events/processors/voice_events.py

```diff
@@ -13,15 +13,19 @@
 class VoiceEvents(EventMixinTemplate):
     @Processor.define()
     async def _on_raw_voice_state_update(self, event: "RawGatewayEvent") -> None:
-        before = copy.copy(self.cache.get_voice_state(event.data["user_id"])) or None
         after = await self.cache.place_voice_state_data(event.data)
```

Comment:
@AstreaTSS This after = `place_voice_state_data` is still sorta a bug if the user is the bot itself, it's setting global state for the bot. I didn't see any methods for like `get_voice_state_data` that fetch the voice state object without updating the global state. I'm not sure this really matters though and I didn't want to touch that code unless it's necessary.

Reply:
That's actually a good point. We do already have branching if statements... it should be easy enough to define two different after behaviors, yeah?

Reply:
yeah it's easy to define two different behaviors but I'm just not sure if there's already a method somewhere that translates `VoiceStateData` to `VoiceState` without actually updating the cache.

Reply:
I don't think so unfortunately 😅.

Now looking at this logic, I do realize that perhaps making `before` in the bot state the thing being updated may not be a great idea. I... suppose you could do something like:

```python
# after still defined as before at the top of the function

# if statement stuff
cached_state = self.cache.get_bot_voice_state(event.data["guild_id"])
before = copy.copy(cached_state)

if cached_state:
    await cached_state._voice_state_update(before, after, event.data)
    after = cached_state
```

I... think that would work? A bit complex, but still.

Reply:
I'm kinda confused, since `_voice_state_update` isn't even using the `before` object, wouldn't it not matter?

I made an update just to add an `update_cache` param to the `place_voice_state_data` method which hopefully should have the least side effect possible.

Reply:
> I'm kinda confused, since `_voice_state_update` isn't even using the `before` object, wouldn't it not matter?

Oops, yeah, forgot about that. Still preferable to copy the state just in case.

Reply:
yuh copied it in the last changed, good call